src/interfaces.ts

```diff
@@ -11,15 +11,26 @@ export interface BatchUpdateDocuments {
   delete: {documentId: string; deleteChildren: boolean}[];
 }
 
+/**
+ * @param {number} remainingDocumentCount Number of remaining documents to upload. This value decreases each time a new batch is uploaded
+ * @param {number} totalDocumentCount Total number documents to upload
+ */
+export interface UploadProgress {
+  remainingDocumentCount: number;
+  totalDocumentCount: number;
+}
+
 /**
  *
  * @param {string[]} files Files from which the documentBuilders were generated
  * @param {DocumentBuilder[]} batch List of the uploaded DocumentBuilders
+ * @param {UploadProgress} progress Progress of the upload process
  * @param {AxiosResponse} res Axios response
  */
 export interface UploadBatchCallbackData {
   files: string[];
   batch: DocumentBuilder[];
+  progress?: UploadProgress;
```

Comment:
Ideally, I would like the progress value to always be returned, but the ways the FileConsumer was designed (with the internal promise), it's hard to do so without a refactor.

Reply:
👶 🐾  : that could be done in another PR